### PR TITLE
NN-3914 springdoc v3

### PR DESCRIPTION
**Pull request by timoj58**

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/BatchController.java [hunk 1]
```diff
@@ -22,7 +24,7 @@
 import static uk.gov.justice.digital.hmpps.keyworker.batch.PrisonStatsRoute.DIRECT_PRISON_STATS;
 import static uk.gov.justice.digital.hmpps.keyworker.batch.UpdateStatusRoute.DIRECT_UPDATE_STATUS;
 
-@Api(tags = {"batch"}, hidden = true)
+@Tag(name = "batch")
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/BatchController.java:27]: Any reason for removing the "hidden" - I know this should never be called by users, just cron jobs

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/dto/AllocationHistoryDto.java [hunk 2]
```diff
@@ -3,6 +3,7 @@
 
 import com.fasterxml.jackson.annotation.JsonInclude;
 import io.swagger.annotations.ApiModelProperty;
```
Comment (sp-watson) [inline src/main/java/uk/gov/justice/digital/hmpps/keyworker/dto/AllocationHistoryDto.java:5]: Is this still used? This comment applies to most of these files

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/BatchController.java (hunk 1 above)
Comment (timoj58) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/BatchController.java:27]: Tag doesnt have a hidden option, all the end points are hidden though, they also have security settings that i assume do not work (you would need to log in to see them).  My assumption is a lof of the api references are cut and paste code, given the differences and un-used things

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/dto/AllocationHistoryDto.java (hunk 2 above)
Comment (timoj58) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/dto/AllocationHistoryDto.java:5]: no i will remove, thought i had deleted, perhaps not.  Was expecting the linter to remove, running reformat / lint again

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/BatchController.java (hunk 1 above)
Comment (sp-watson) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/controllers/BatchController.java:27]: yeah, ok

File: src/main/java/uk/gov/justice/digital/hmpps/keyworker/dto/AllocationHistoryDto.java (hunk 2 above)
Comment (sp-watson) [reply src/main/java/uk/gov/justice/digital/hmpps/keyworker/dto/AllocationHistoryDto.java]: hmm.. you are right - the linter should have picked it up. This is one of the first APIs so i think it hasn't got the linter set up properly. That would be worth investigating some time.

**Review by sp-watson — APPROVED** [review]

**Pull request merged**